clj-refactor.el

```diff
@@ -1087,19 +1087,22 @@ If CLJS? is T we insert in the cljs part of the ns declaration."
 
 (defun cljr--cljs-file-p (&optional buf)
   "Is BUF, or the current buffer, visiting a cljs file?"
-  (string-equal (file-name-extension (buffer-file-name (or buf (current-buffer))))
-                "cljs"))
+  (when-let ((bfn (buffer-file-name (or buf (current-buffer)))))
+    (string-equal (file-name-extension bfn)
+                  "cljs")))
 
 (defun cljr--cljc-file-p (&optional buf)
   "Is BUF, or the current buffer, visiting a cljc file?"
-  (string-equal (file-name-extension (buffer-file-name (or buf (current-buffer))))
-                "cljc"))
+  (when-let ((bfn (buffer-file-name (or buf (current-buffer)))))
+    (string-equal (file-name-extension bfn)
+                  "cljc")))
 
 (defun cljr--clj-file-p (&optional buf)
   "Is BUF, or the current buffer, visiting a clj file?"
   (or (eq major-mode 'clojure-mode)
-      (string-equal (file-name-extension (buffer-file-name (or buf (current-buffer))))
-                    "clj")))
+      (when-let ((bfn (buffer-file-name (or buf (current-buffer)))))
+        (string-equal (file-name-extension bfn)
+                      "clj"))))
```

Comment:
Maybe we should extract this logic into a single `(cljr--file-type-p extension buffer)`? and re-use this helper for each of the cljc--file-p, clj-file-p, cljs-file-p?

Separately though, I don't think I noticed that major-mode `clojure-mode is sufficient for clj-file-p. That seems ... questionable if the file is cljs or cljc?

Reply:
CIDER uses this sort of logic:

```lisp
(cond
     ((derived-mode-p 'clojurescript-mode) ,,,)
     ((derived-mode-p 'clojurec-mode) ,,,)
     ((derived-mode-p 'clojure-mode) ,,,)
```

i.e. the clojure-mode mode is only to be expected for .clj files.

We could start using derived-mode-p though